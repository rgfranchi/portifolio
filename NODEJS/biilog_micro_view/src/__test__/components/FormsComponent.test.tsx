import React from "react";

import { cleanup } from "@testing-library/react";

import Enzime, { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";

import {
  FromSelect,
  FormDatePicker,
  StringArrayToList,
} from "../../components/FormsComponent";

afterEach(cleanup);

Enzime.configure({ adapter: new Adapter() });

describe("FormsComponent snapshot", () => {
  test("StringArrayToList", () => {
    const wrapper = shallow(
      <StringArrayToList stringArray={["texto 1", "texto 2", "texto 3"]} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test("FromSelect", () => {
    const wrapper = mockFromSelect();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test("FormDatePicker", () => {
    const wrapper = shallow(
      <FormDatePicker
        name="mockDatePicker"
        formikSetFieldValue={() => {}}
        value="2020-01-01"
        dateFormat="dd/MM/yyyy"
        format="yyyy-MM-dd"
        isInvalid={true}
        showTimeInput={true}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe("FormsComponent execute", () => {
  test("FromSelect", () => {
    const wrapper = mockFromSelect();
    let formSelect = wrapper.find("FormControl");
    formSelect.simulate("change", { value: ["VALOR 3"] });
    expect(mockFromChange).toHaveBeenCalledWith({ value: ["VALOR 3"] });
  });
});

const mockFromChange = jest.fn();
const mockFromData = [
  { id: 1, value: "VALOR 1" },
  { id: 2, value: "VALOR 2" },
  { id: 3, value: "VALOR 3" },
];
const mockFromSelect = (
  dataList: any = mockFromData,
  handleChange: any = mockFromChange
) => {
  return shallow(
    <FromSelect
      label="Teste Select Mock"
      name="mockSelect"
      dataId={"2"}
      dataList={dataList}
      dataEmpty={{ id: 0, value: "VALOR 0" }}
      handleChange={handleChange}
      handleBlur={() => {}}
      errors={""}
      isInvalid={true}
    />
  );
};
