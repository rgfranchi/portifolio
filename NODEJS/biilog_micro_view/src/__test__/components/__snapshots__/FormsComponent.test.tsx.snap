// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormsComponent snapshot FormDatePicker 1`] = `
<FormControl
  as={[Function]}
  dateFormat="dd/MM/yyyy"
  isInvalid={true}
  name="mockDatePicker"
  onChange={[Function]}
  selected={2020-01-01T00:00:00.000Z}
  showTimeInput={true}
/>
`;

exports[`FormsComponent snapshot FromSelect 1`] = `
<FormGroup
  controlId="ctrlCompany"
>
  <FormLabel
    column={false}
    srOnly={false}
  >
    Teste Select Mock
  </FormLabel>
  <FormControl
    as="select"
    isInvalid={true}
    name="mockSelect"
    onBlur={[Function]}
    onChange={[MockFunction]}
    value="2"
  >
    <option
      value={0}
    >
      VALOR 0
    </option>
    <option
      key="0"
      selected={false}
      value={1}
    >
      VALOR 1
    </option>
    <option
      key="1"
      selected={true}
      value={2}
    >
      VALOR 2
    </option>
    <option
      key="2"
      selected={false}
      value={3}
    >
      VALOR 3
    </option>
  </FormControl>
  <Feedback
    type="invalid"
  />
</FormGroup>
`;

exports[`FormsComponent snapshot StringArrayToList 1`] = `
<ul>
  <li
    key="0"
  >
    texto 1
  </li>
  <li
    key="1"
  >
    texto 2
  </li>
  <li
    key="2"
  >
    texto 3
  </li>
</ul>
`;
